# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/notification/NotificationParent.cpp
# Commit: bda570180adf
# Full Hash: bda570180adfcc20891d5cfb47406756fe8903ad
# Author: Kagami Sascha Rosylight <saschanaz@outlook.com>
# Date: 2024-11-12 09:32:50
# Regressor Bug: 1928702
# File Overlap Count: 1
# Description:
#   Bug 1928702 - Part 11: Migrate ShowPersistentNotification to use IPC r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D227696
# ==============================================================================

diff -r c942992fa981 -r bda570180adf dom/notification/NotificationParent.cpp
--- a/dom/notification/NotificationParent.cpp	Mon Nov 11 20:34:15 2024 +0000
+++ b/dom/notification/NotificationParent.cpp	Mon Nov 11 20:34:15 2024 +0000
@@ -62,6 +62,9 @@
     // XXX: QM_TRY?
     (void)NS_WARN_IF(NS_FAILED(UnpersistNotification(mPrincipal, mId)));
     (void)NS_WARN_IF(NS_FAILED(FireCloseEvent()));
+
+    // Unpersisted already and being unregistered already by nsIAlertsService
+    mDangling = true;
     Close();
 
     return NS_OK;